feature_engine/discretisation/base_discretiser.py

```diff
@@ -0,0 +1,120 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
```

Comment:
good call creating this base class.

To summarize, the base class should have a common init shared by most discretisers and a common transform method, also shared by most discretisers.